### PR TITLE
Moved over the `UrlReaderService` default implementation to `backend-defaults`

### LINKED ISSUE #25004: [Url Reader] Prefix all `ReadTree` types with the `URLReader` service name

### LINKED ISSUE #25005: [Url Reader] Prefix `SearchOptions` with the `UrlReader` service name

### PR DESCRIPTION
**Pull request by freben**
Closes #25004
Closes #25005

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
...
     "build:api-reports": "yarn build:api-reports:only --tsc",
-    "build:api-reports:only": "backstage-repo-tools api-reports --allow-warnings 'packages/core-components,plugins/+(catalog|catalog-import|git-release-manager|jenkins|kubernetes)' -o ae-wrong-input-file-type --validate-release-tags",
+    "build:api-reports:only": "backstage-repo-tools api-reports --allow-warnings 'packages/backend-common,packages/core-components,plugins/+(catalog|catalog-import|git-release-manager|jenkins|kubernetes)' -o ae-wrong-input-file-type --validate-release-tags",
```
Comment (freben) [inline package.json:20]: ⚠️ 

well this package is going away anyway, but i couldn't make its api report stop complaining

File: packages/backend-defaults/package.json [hunk 2]
```diff
@@ -82,38 +82,60 @@
     "test": "backstage-cli package test"
   },
   "dependencies": {
+    "@aws-sdk/abort-controller": "^3.347.0",
+    "@aws-sdk/client-codecommit": "^3.350.0",
+    "@aws-sdk/client-s3": "^3.350.0",
+    "@aws-sdk/credential-providers": "^3.350.0",
+    "@aws-sdk/types": "^3.347.0",
```
Comment (Rugvip) [inline packages/backend-defaults/package.json:89]: Well now I really wanna be able to ship this as a service module :p

File: packages/backend-defaults/src/entrypoints/urlReader/lib/GoogleGcsUrlReader.ts [hunk 3]
```diff
@@ -56,24 +56,24 @@ const parseURL = (
...
     );
-    let storage: Storage;
+    let storage: GoogleCloud.Storage;
     if (!gcsConfig.clientEmail || !gcsConfig.privateKey) {
       logger.info(
         'googleGcs credentials not found in config. Using default credentials provider.',
       );
-      storage = new Storage({
+      storage = new GoogleCloud.Storage({
         userAgent: `backstage/backend-common.GoogleGcsUrlReader/${packageinfo.version}`,
```
Comment (Rugvip) [inline packages/backend-defaults/src/entrypoints/urlReader/lib/GoogleGcsUrlReader.ts:73]: Should prolly update this while at it, to `backend-defaults`, same below

**Review by Rugvip — APPROVED** [review]
Nice! 🎉 

Thinking it would be nice to have this merged before the release tomorrow, just some tiny nit

**Review by camilaibs — APPROVED** [review]
Thanks 🙌🏻

File: packages/backend-defaults/package.json (hunk 2 above)
Comment (freben) [reply packages/backend-defaults/package.json:89]: Aye.

File: packages/backend-defaults/src/entrypoints/urlReader/lib/GoogleGcsUrlReader.ts (hunk 3 above)
Comment (freben) [reply packages/backend-defaults/src/entrypoints/urlReader/lib/GoogleGcsUrlReader.ts]: I actually thought, what if they are used somewhere :) But yeah OK I changed them now

**Pull request merged**